### PR TITLE
vlang 0.3

### PR DESCRIPTION
**Pull request by Porkepix**
Created with `brew bump-formula-pr`.

`resource` blocks may require updates.

### CODE REVIEW

File: Formula/vlang.rb [hunk 1]
```diff
@@ -2,8 +2,8 @@ class Vlang < Formula
   desc "V programming language"
   homepage "https://vlang.io"
   # NOTE: Keep this in sync with V compiler below when updating
```
Comment (carlocab) [inline Formula/vlang.rb:4]: The `resource` probably needs updating.

**Review by gromgit — CHANGES_REQUESTED** [review]

File: Formula/vlang.rb [hunk 2]
```diff
@@ -26,7 +26,7 @@ class Vlang < Formula
     # "[v:master] {short SHA of the vlang release commit} - {vlang version number}".
     # The sources of this V compiler commit need to be used here
     url "https://github.com/vlang/vc.git",
-        revision: "fd5f57740ff6d7a8566b774318df54c2fa460f92"
+        revision: "5dc8384638f515a098f75296fa7123c5e6a4964a"
   end
```
[the remark this answers is not in the document]
Comment (gromgit) [reply Formula/vlang.rb:30]: Ah, it's pulling HEAD. Never mind then.

[the remark this answers is not in the document]
Comment (carlocab) [reply Formula/vlang.rb:30]: We could pretty easily mimic what the `Makefile` does while keeping the resource, though.

[the remark this answers is not in the document]
Comment (gromgit) [reply Formula/vlang.rb:30]: In that case...
```
	$(CC) $(CFLAGS) -std=gnu11 -w -o v1 vc/v.c -lm -lexecinfo -lpthread $(LDFLAGS)
```
@Porkepix , try adding `-lexecinfo` and `-lpthread` to the resource build.

The rest looks like a rebuild of `v` (which `v self` also does), and an "is `tcc` present?" advisory script (which is pointless for CI).

[the remark this answers is not in the document]
Comment (carlocab) [reply Formula/vlang.rb:30]: Actually, it looks like `v` is being built with itself twice. is that what `v self` does?

[the remark this answers is not in the document]
Comment (gromgit) [reply Formula/vlang.rb:30]: Based on https://github.com/vlang/v/blob/d679146a8003be6dc21f5b677753fd1e7403faa3/cmd/tools/vself.v, it seems `v self` only rebuilds itself once. I'm guessing that the Makefile double-build is just meant to test their `-no-parallel` flag, not sure if it's worth bothering about.

[the remark this answers is not in the document]
Comment (carlocab) [reply Formula/vlang.rb:30]: This is actually a known pattern for optimising compiler builds, so I suspect that's what they're doing here.

See https://llvm.org/docs/AdvancedBuilds.html for an example.

File: Formula/vlang.rb [hunk 3]
```diff
@@ -26,14 +26,21 @@ class Vlang < Formula
...
 
   def install
+    # Check build commands in the `Makefile`. We don't use `make`
+    # to avoid building from arbitrary commits in the `vc` repo.
     resource("vc").stage do
-      system ENV.cc, "-std=gnu11", "-w", "-o", buildpath/"v", "v.c", "-lm"
+      bootstrap_libs = %w[-lm -lpthread]
+      bootstrap_libs.insert(1, "-lexecinfo") unless OS.mac?
```
Comment (carlocab) [inline Formula/vlang.rb:37]:
```suggestion
```

File: Formula/vlang.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (carlocab) [reply Formula/vlang.rb:30]: Actually, I just realised that we should be looking at the [`GNUMakefile`](https://github.com/vlang/v/blob/master/GNUmakefile), and not the `Makefile`. It, however, seems to do [a similar thing](https://github.com/vlang/v/blob/584c03512027c2c526e3825e3b83f04063960707/GNUmakefile#L120-L122):

```make
fresh_vc:
	rm -rf $(VC)
	$(GITFASTCLONE) $(VCREPO) $(VC)
```

File: Formula/vlang.rb [hunk 4]
```diff
@@ -26,14 +26,15 @@ class Vlang < Formula
...
   end
 
   def install
-    resource("vc").stage do
-      system ENV.cc, "-std=gnu11", "-w", "-o", buildpath/"v", "v.c", "-lm"
-    end
-    system "./v", "self"
+    vc = buildpath/"vc"
+    vc.install resource("vc")
+    (buildpath/"thirdparty/tcc/tcc.exe").write_exec_script ENV.cc
```
Comment (carlocab) [inline Formula/vlang.rb:35]: Looks like this doesn't work. Should we just add the `tccbin` resource here? I feel like there should be a way not to have to...

**Comment by harens** [comment]
Hi 👋 I've been maintaining weekly builds of V over at MacPorts for a while now. A lot of commits have gone into getting it to build successfully and reproducibly, so I thought you might find our annotated [build script](https://github.com/macports/macports-ports/blob/master/lang/vlang/Portfile) and macOS 11 arm [log file](https://build.macports.org/builders/ports-11_arm64-builder/builds/61481/steps/install-port/logs/stdio) useful.

> Vlang needs libgc.a from TCC by default.

The issue is that V hard codes a lot of prefixes by default. I understand that Homebrew is much more strict on patches than MacPorts, but you mind find my set of [patches](https://github.com/macports/macports-ports/tree/master/lang/vlang/files) useful. I hope it helps.

**Comment by carlocab** [comment]
Thanks for dropping by, @harens. That does help a lot.

**Review by eelcocramer — COMMENTED** [review]
Never mind. It seems that my build script after installation is not able to build v code.

**Comment by chenrui333** [comment]
vlang 0.3.2 is avail, I am gonna give that a shot in my local today.

**Comment by chenrui333** [comment]
#119181

**Pull request closed without merging**